components/net_traits/response.rs

```diff
@@ -72,6 +75,55 @@ pub enum ResponseMsg {
     Errored
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize, HeapSizeOf)]
+pub enum ResponseError {
+    ConnectionError(String),  // FIMXE: don't use string
```

Comment:
I'd love to make this `ConnectionError(std::io::Error)`, but can't seem to get it to work right. Any other ideas here?